argilla-server/src/argilla_server/api/errors/v1/exception_handlers.py

```diff
@@ -75,3 +75,7 @@ async def update_distribution_with_existing_responses(request, exc):
             status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
             content={"code": exc.code, "message": exc.message},
         )
+
+    @app.exception_handler(errors.AuthenticationError)
+    async def authentication_error(request, exc):
```

Comment:
I would move this exception handler to be the first one (with the rest of generic errors) following the order of the status code, in this case `401`.